src/dbnode/storage/bootstrap/bootstrapper/fs/source.go

```diff
@@ -646,28 +651,13 @@ func (s *fileSystemSource) readNextEntryAndRecordBlock(
 		return fmt.Errorf("error reading data file: %v", err)
 	}
 
-	var (
-		entry  result.DatabaseSeriesBlocks
-		tags   ident.Tags
-		exists bool
-	)
-	runResult.Lock()
-	defer runResult.Unlock()
-
-	entry, exists = shardResult.AllSeries().Get(id)
-	if exists {
-		// NB(r): In the case the series is already inserted
-		// we can avoid holding onto this ID and use the already
-		// allocated ID.
-		id.Finalize()
-		id = entry.ID
-		tags = entry.Tags
-	} else {
-		tags, err = convert.TagsFromTagsIter(id, tagsIter, s.idPool)
-		if err != nil {
-			return fmt.Errorf("unable to decode tags: %v", err)
-		}
+	ref, err := accumulator.CheckoutSeries(id, tagsIter)
```

Comment:
TODO: Need to add runResult.Lock() and defer runResult.Unlock() since accumulator cannot be used in parallel.